test/Test/Pos/Block/LogicSpec.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Pos.Block.LogicSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import           Data.Default               (def)
+import           Serokell.Data.Memory.Units (Byte, fromBytes)
+import           Test.Hspec                 (Spec, describe, runIO)
+import           Test.Hspec.QuickCheck      (prop)
+import           Test.QuickCheck            (Gen, Property, Testable, arbitrary, choose,
+                                             counterexample, elements, forAll, generate,
+                                             listOf, listOf1, oneof, property)
+
+import           Pos.Binary.Class           (biSize)
+import qualified Pos.Binary.Class           as Bi
+import           Pos.Block.Arbitrary        ()
+import           Pos.Block.Logic            (createMainBlockPure)
+import qualified Pos.Communication          ()
+import           Pos.Constants              (blkSecurityParam, genesisMaxBlockSize)
+import           Pos.Core                   (SlotId (..))
+import           Pos.Crypto                 (SecretKey)
+import           Pos.Ssc.Class              (Ssc (..), sscDefaultPayload)
+import           Pos.Ssc.GodTossing         (GtPayload (..), SscGodTossing,
+                                             commitmentMapEpochGen, mkVssCertificatesMap,
+                                             vssCertificateEpochGen)
+import           Pos.Txp.Core               (TxAux)
+import           Pos.Types                  (BlockHeader, MainBlock, ProxySKEither,
+                                             ProxySKHeavy, SmallGoodTx (..),
+                                             goodTxToTxAux)
+import           Pos.Update.Core            (UpdatePayload (..))
+import           Pos.Util.Arbitrary         (makeSmall)
+
+
+spec :: Spec
+spec = describe "Block.Logic" $ do
+
+    -- Sampling the minimum empty block size
+    (sk0,prevHeader0) <- runIO $ generate arbitrary
+
+    -- We multiply by 1.5 because there are different possible
+    -- combinations of empty constructors and there's no out-of-box
+    -- way to get maximum of them. Some settings produce 390b empty
+    -- block, some -- 431b.
+    let emptyBSize0 :: Byte
+        emptyBSize0 = biSize (noSscBlock infLimit [] prevHeader0 sk0 def []) -- in bytes
+        emptyBSize :: Integral n => n
+        emptyBSize = round $ (1.5 * fromIntegral emptyBSize0 :: Double)
+
+    describe "createMainBlockPure" $ do
+        prop "empty block size is sane" $ emptyBlk $ \blk ->
```

Comment:
Here and afterwards, where possible, can you separate the test tree and the actual tests?
I think it should be

```
spec :: Spec
spec = describe ... 
    prop desc1 test1
    prop desc2 test2

test1 :: ...
test1 = ...

test2 :: ...
test2 = ...
```

Right, you inline the tests into the `prop` line, and reduces clarity.